### DISCUSSION TITLE
Why depend on data-domain attribute instead of using location.host inside the script?

### DISCUSSION THREAD

**Discussion by srameshr** [plausible/analytics#590] [Anything Else]
Our host does not accept any data attribute on the script tag. Can we change the below line that is https://github.com/plausible/tracker/blob/master/src/plausible.js#L8 from 
```
  var scriptEl = document.querySelector('[src*="' + plausibleHost +'"]')
  var domain = scriptEl && scriptEl.getAttribute('data-domain')
```

to

```
  var scriptEl = document.querySelector('[src*="' + plausibleHost +'"]')
  var domain = location.hostname
```

**Comment by ukutaht** [comment]
It used to work like that but there were many requests to decouple the site id in plausible from the actual hostname of the website. For example, if you have a website with different languages for subdomains, you want `es.example.com`, `de.example.com` and `en.example.com` to all count towards pageviews on `example.com`. Therefore overriding the hostname is required.

**Comment by srameshr** [comment]
But does a fork replacing data attribute with hostname raise any bugs? I ask this because I am planning to host the plausible.js in our cdn but with our changes on top of it.

**Reply by ukutaht** [reply]
I don't see any problems with that

**Reply by srameshr** [reply]
Thank you!

**Comment by sandstrom** [comment]
Being able to customize has several use-cases. I hope that's not removed.
